lib/puppet/application/agent.rb

```diff
@@ -358,13 +358,21 @@ def run_command
   end
 
   def fingerprint
-    sm = Puppet::SSL::StateMachine.new(onetime: true)
-    ssl_context = sm.ensure_client_certificate
-    if ssl_context
-      puts Puppet::SSL::Digest.new(options[:digest].to_s, ssl_context.client_cert.to_der).to_s
+    cert_provider = Puppet::X509::CertProvider.new
+    client_cert = cert_provider.load_client_cert(Puppet[:certname])
+    if client_cert
+      puts Puppet::SSL::Digest.new(options[:digest].to_s, client_cert.to_der).to_s
+    else
+      csr = cert_provider.load_request(Puppet[:certname])
+      if csr
+        puts Puppet::SSL::Digest.new(options[:digest].to_s, csr.to_der).to_s
+      else
+        $stderr.puts _("Fingerprint asked but no certificate nor certificate request have yet been issued")
```

Comment:
Should this be "Fingerprint asked but neither the certificate, nor the certificate request have been issued"? The "no ... nor" pattern feels awkward there.

...and after saying that I noticed that it was already using that wording down further, so feel free to ignore, or not.

Reply:
Agree `no ... nor` is awkward. I'll update that.